### ISSUE TITLE
Internal error function createSimCode failed

### ISSUE THREAD

**Issue by sbaetzing** [OpenModelica/OpenModelica#7485]
### Description
Translation of the below mentioned models end ups wwith the following error:
`Internal error function createSimCode failed [Transformation from optimised DAE to simulation code structure failed]`
Not sure why this error occurs. Earlier versions were already able to translate the models. 

### Steps to Reproduce

Run one of the following testcases in the Bosch Rexroth Simulation Libraries:
Rexroth_Generic_Test.Hydraulics.Examples.InitializationConcept
Rexroth_Generic_Test.Hydraulics.DirectionalValves.DirectionalValve33_Easy_TC01
Rexroth_Generic_Test.Hydraulics.DirectionalValves.DirectionalValve43_Easy_TC01
Rexroth_Generic_Test.Hydraulics.PropValvesValves.PropValvesValve33_Easy_TC01
Rexroth_Generic_Test.Hydraulics.PropValvesValves.PropValvesValve43_Easy_TC01
Rexroth_Generic_Test.Hydraulics.PropValvesValves.PropValvesValve43_Easy_TC02

### Expected Behavior

Testcases should simulate successful

### Screenshots
SOrry i cant upload any pictures with any browsers. Not sure why it is not working.

### Version and OS

- OpenModelica v1.18.0-dev-280-g0b76908279 (64-bit)
- OMSimulator v2.1.1.post70-gdf73447-mingw
- OS: Windows 10, 64 bit
 - Versions of used Modelica libraries if applicable: MSL 3.2.3, Bosch Rexroth Simulation Libraries (1.4)

### Additional Context
You should already have the Bosch Rexroth Simulation Libraries (1.4)
 aswell as all the correpsonding test libraries. If not Niklas Worchesch can upload it again.

Best wishes

Sven Bätzing

**Comment by perost** [comment]
It fails in `SimCodeUtil.addVarToArrayIndexMapping` on `propValve43_Easy.flowTableAT[2,1]` (after succeeding on `propValve43_Easy.flowTableAT[1,1]`), where `propValve43_Easy.flowTableAT` has not been scalarized in the frontend because it has a binding equation consisting of a function call returning an array.

**Comment by casella** [comment]
@sbaetzing, please check with the next nightly build if this issue is solved for you, and in that case please close the ticket.

**Comment by niklwors** [comment]
@kabdelhak @casella Hello, unfortunately I still got the same errors for these models.

**Comment by casella** [comment]
@niklwors are you sure that @kabdelhak's commit is in the build you are using?

**Comment by niklwors** [comment]
@casella  @kabdelhak yes, i used the version from 24.09.21

**Comment by mahge** [comment]
There is a chance this could have been fixed by #8019. Per's description is 

> has not been scalarized in the frontend because it has a binding equation consisting of a function call returning an array.

Which is exactly what #8019 tried to improve in the Backend. @niklwors Can you give it another shot and see if anything changed?

**Comment by mahge** [comment]
I have checked all the models listed in the description and all of them compile fine after #8019. Some fail during simulation but that is another issue. So I am going to close this. Please open it if the issue is still there from your side.

**Comment by niklwors** [comment]
There is now the problem with wrong simulation results for the test cases . The issue is, the  an array assignment went wrong. e.g.
in the DirectValve component from test case ```DirectionalValve33_Easy_TC01``` the assignment  
```parameter Real[:,1] flowTableAT=   Rexroth_Generic.Hydraulics.Valves.Internal.ExtractTable(qNTableEasy, false)```  results in an array with zero elements which is not correct.

**Comment by casella** [comment]
@mahge, this is for you to check.
